packages/gateway-trpc/src/crypto.ts

```diff
@@ -0,0 +1,26 @@
+import { createRemoteJWKSet, jwtVerify } from "jose"
+
+export class JwtService {
+  private jwks: ReturnType<typeof createRemoteJWKSet> | null = null
+  private readonly jwksUrl: URL
+
+  public constructor(
+    private readonly issuer: string,
+    private readonly audiences: string[]
+  ) {
+    const issuerWithoutTrailingSlash = issuer.replace(/\/$/, "")
+    this.jwksUrl = new URL(`${issuerWithoutTrailingSlash}/.well-known/jwks.json`)
+  }
+
+  public async verify(accessToken: string) {
+    this.jwks ??= createRemoteJWKSet(this.jwksUrl)
```

Comment:
I think that key might rotate, and has a cache-control-setting limiting how long we should cache it

if this is specifically from `https://auth.online.ntnu.no/.well-known/jwks.json` it has `Cache-Control` `public, max-age=15, stale-while-revalidate=15, stale-if-error=86400`, so we can cache it for 30s at a time

Reply:
Right can look into that, I'm used to cognito which doesn't rotate the jwks at all :upside_down_face:

Reply:
Fixed in 44d5953c